### PR TITLE
Install tools base layout

### PR DESCRIPTION
**Pull request by Jlisowskyy**
Things done in this PR:
- installer for cross-compiler GCC
- basic layout for install scripts
- basic helpers functions

### CODE REVIEW

**Review by F1r3D3v — APPROVED** [review]
LGTM

File: env/helpers.bash [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+dump_error() {
+  help
+
+  pretty_error "$1"
+  exit 1
+}
+
+assert_always() {
+  dump_error "ASSERT failed in ${BASH_SOURCE[1]} at line ${BASH_LINENO[0]}"
```
Comment (kryczkal) [inline env/helpers.bash:16]: How is this different from dump_error? Maybe merge the two

File: env/build_cross_compile.bash [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+  build_gdb
+  build_gcc
+
+  add_to_user_path
+
+  pretty_success "Build: ${CROSS_COMPILE_BUILD_SCRIPT_PATH} completed successfully"
+}
+
+parse_args() {
+  while [[ $# -gt 0 ]]; do
```
Comment (kryczkal) [inline env/build_cross_compile.bash:214]: No option to parse quiet mode that already exists as argument to runner

File: env/build_cross_compile.bash [hunk 3]
```diff
@@ -0,0 +1,271 @@
...
+  pretty_success "GDB built correctly"
+
+  pretty_info "Installing GDB"
+  runner "Failed to install GDB" make install-gdb
+  pretty_success "GDB installed correctly"
+
+  pretty_success "GDB build completed"
+}
+
+check_path() {
```
Comment (kryczkal) [inline env/build_cross_compile.bash:166]: I'm sure this functionality will be sometime later required in other places of code. Can be potentially moved to helpers and get the "new_path" as argument.

suggestion - check_is_in_path?

File: env/build_cross_compile.bash [hunk 4]
```diff
@@ -0,0 +1,271 @@
...
+            CROSS_COMPILE_BUILD_IS_ADDED_TO_PATH=true
+            return 0
+        fi
+    done
+
+    pretty_info "Path: ${new_path} does not exist in PATH"
+}
+
+add_to_user_path() {
+  local new_path="${CROSS_COMPILE_BUILD_TOOL_DIR}/bin"
```
Comment (kryczkal) [inline env/build_cross_compile.bash:186]: Same as check_path - can be moved to helpers and get new_path as arg

File: env/pretty_print.bash [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+    local red='\033[0;31m'
+    local yellow='\033[1;33m'
+    local green='\033[0;32m'
+    local blue='\033[0;34m'
+    local cyan='\033[0;36m'
+    local nc='\033[0m'
+    local bold='\033[1m'
+
+    case "${type,,}" in
```
Comment (kryczkal) [inline env/pretty_print.bash:16]: Maybe add a \t between the label and message for consistent spacing in outputs?

**Review by kryczkal — CHANGES_REQUESTED** [review]
I gave minor remarks that I think will make this even better. 
Otherwise - Great job!

File: env/helpers.bash [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+}
+
+assert_no_argument() {
```
[the remark this answers is not in the document]
Comment (Jlisowskyy) [reply env/helpers.bash:19]: It should be named like that from the beginning, fixed.

File: env/helpers.bash (hunk 1 above)
Comment (Jlisowskyy) [reply env/helpers.bash:16]: Dump error is simple exiting function with some log message and maybe be used in all cases, assert fails when something unexpected happens, something what should never happen during execution.

File: env/build_cross_compile.bash (hunk 2 above)
Comment (Jlisowskyy) [reply env/build_cross_compile.bash:214]: There was quiet argument for runner and verbose for CLI, fixed all of them now uses verbose

File: env/build_cross_compile.bash (hunk 3 above)
Comment (Jlisowskyy) [reply env/build_cross_compile.bash:166]: Sure, great suggestion

File: env/build_cross_compile.bash (hunk 4 above)
Comment (Jlisowskyy) [reply env/build_cross_compile.bash:186]: Already done

File: env/pretty_print.bash (hunk 5 above)
Comment (Jlisowskyy) [reply env/pretty_print.bash:16]: looks better now, done

Comment (Jlisowskyy) [reply env/pretty_print.bash:16]: image

**Review by kryczkal — APPROVED** [review]

**Pull request merged**